### ISSUE TITLE
[BUG] HTTP spans have wrong parent in ApplicationInsights

### ISSUE THREAD

**Issue by lmolkova** [Azure/azure-sdk-for-java#22272]
**Note** this is likely azure-core issue, not related to storage

**Describe the bug**
HTTP span is reported as child of incoming request, not the logical downloadBlob in SpringBoot app

Here is the [transaction](https://ms.portal.azure.com/#blade/AppInsightsExtension/DetailsV2Blade/DataModel/%7B%22eventId%22%3A%222d1e21bf-cbf0-11eb-869c-0050f20fa676%22%2C%22timestamp%22%3A%222021-06-13T02%3A35%3A52.647Z%22%7D/ComponentId/%7B%22Name%22%3A%22spring-boot-1623527233948%22%2C%22ResourceGroup%22%3A%22limolkova%22%2C%22SubscriptionId%22%3A%22faa080af-c1d8-40ad-9cce-e1a450ca5b57%22%7D)

***Exception or Stack Trace***
N/A

**To Reproduce**
1. create a simple SpringBoot app
2. in the controller make a call to upload blob
3. deploy to azure app service
3. configure app insights on the app-service
4. [enable Azure SDK collection](https://docs.microsoft.com/en-us/azure/azure-monitor/app/java-standalone-config#auto-collected-azure-sdk-telemetry-preview)

***Code Snippet***
```java
	@RequestMapping("/")
		BlobClient blobClient = this.containerClient.getBlobClient("blob" + System.currentTimeMillis());
		blobClient.upload(BinaryData.fromString("test"));

		return "Greetings from Spring Boot!";
	}

```

**Expected behavior**
http span (`azmondemo`) should be child of `upload blob` span

**Screenshots**
![image](https://user-images.githubusercontent.com/2347409/121919294-bb7f6000-cceb-11eb-8fa1-d9ec8b1a16c7.png)

**Setup (please complete the following information):**
 - OS: Linux
 - IDE: IntelliJ
 - Library/Libraries: `azure-core:1.17.0`

**Additional context**

**Information Checklist**
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by lmolkova** [comment]
Investigation:
- Azure Core produces spans as expected.
- Bytecode instrumentation agent adds another HTTP span 
- ApplicationInsights SDK suppresses Azure SDK Http span (which is a duplicate of above)

Here's how it looks in Zipkin

![image](https://user-images.githubusercontent.com/2347409/122272334-4a7bac00-ce95-11eb-86d9-375d6ca62d71.png)


I.e. all clients who use AppInsights agent are affected, all OTel users too.

Action items
[ ] - long term: OTel needs solution for double-instrumentation across all languages https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/903 
[ ] - short term: I'm looking into possible mitigation for OTel and Application Insights 
[ ] - (minor) investigate span names difference (and other) between Azure SDK and agent instrumentation

 /cc @trask

**Comment by lmolkova** [comment]
ApplicationInsights agent issue does not affect new agent 3.1.1+ if there is no azure-core-tracing-opentelemetry.

If user installs it  AND [enables otel api support](https://github.com/microsoft/ApplicationInsights-Java/wiki/OpenTelemetry-API-support-(3.0)), there is duplication 

![image](https://user-images.githubusercontent.com/2347409/126585644-0a294ead-574d-49b5-822d-27cf78f49f33.png)

**Comment by lmolkova** [comment]
Fixed in [OpenTelemetry Java agent v1.12.0](https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/tag/v1.12.0)
